buildSrc/src/main/java/com/google/firebase/gradle/plugins/FireEscapeArtifactPlugin.kt

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import java.io.File
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.bundling.Zip
+
+class FireEscapeArtifactPlugin : Plugin<Project> {
+  var project: Project? = null
+
+  override fun apply(target: Project) {
+    project = target
+
+    project!!.afterEvaluate {
+      val firebaseLibrary = extensions.findByType(FirebaseLibraryExtension::class.java)
```

Comment:
`findByType` returns null if not found, and below I see you circumvent that by making a null check. This would throw an NPE- which is generally not desired. Instead, you should use `getByType`, which will throw an `UnknownDomainObjectException` on the spot- which would be better for debugging and more null safe.

Additionally, *almost* every time a method requires a class argument, if the method was rewritten in Kotlin, it's likely to be `reified`. This means you can pass the type directly instead of as a parameter, like so:
```kotlin
val firebaseLibrary = extensions.getByType<FirebaseLibraryExtension>()
```